### PR TITLE
feat(react): handle missed enduser messages when connectivty is regained

### PR DESCRIPTION
**Pull request by vanbasten17**
_Remember to tag the PR with **WIP** tag if it's not ready to be merged_.  
[Reference](https://blog.codeminer42.com/on-writing-a-great-pull-request-37c60ce6f31d/)

## Description
* Added functionality to support missed ~messages~ user inputs from enduser.
* Enduser messages are displayed with less opacity until they are correctly delivered to the server.

## Context
When enduser ~messages~ inputs were not delivered correctly, we weren't trying to resend them from `Botonic`-> `Hubtype`. The purpose of this PR is to have a very first approach of this system that sets the basis for keep improving these functionalities.

`resendUnsentInputs` will be triggered:
* after hubtype service has been initialized
* after webchat has been opened
* if an online connection has been detected with `useNetwork` hook.

## Approach taken / Explain the design
* Make use of a `useNetwork` [hook](https://medium.com/the-non-traditional-developer/checking-the-network-connection-with-a-react-hook-ec3d8e4de4ec) to detect network connectivity. By default, the enduser messages (prod) will be initialized with `ack` set to `0` (not delivered), once we receive the response from the server, we update these corresponding messages.
* `resendUnsentUserInputs` will be called every time we regain connectivity or every time the webchat is opened, trying to resend all the `messagesJSON` with `ack:0`, updating them with `ack: 1`

## To documentate / Usage example
**Message to be delivered**
Screenshot 2020-06-25 at 13 30 14

**Message delivered**
Screenshot 2020-06-25 at 13 30 31

**Connectivity issues** (messages wil be dequeued and processed in the same order once we regain connection)
Screenshot 2020-06-25 at 15 38 44

## Testing

The pull request...

- [ ] has unit tests
- [ ] has integration tests
- [ ] doesn't need tests because... **[provide a description]**

### CODE REVIEW

File: packages/botonic-react/src/components/message.jsx [hunk 1]
```diff
@@ -132,6 +139,14 @@ export const Message = props => {
 
   const timestampFormat = timestamps && getTimestampFormat()
 
+  let ack = 0
```
[the remark this answers is not in the document]
Comment (vanbasten17) [reply packages/botonic-react/src/components/message.jsx:142]: sure, we'll take a look on it!

File: packages/botonic-react/src/webchat/messages-reducer.js [hunk 2]
```diff
@@ -38,15 +38,35 @@ export const messagesReducer = (state, action) => {
 
 function updateMessageReducer(state, action) {
   const msgIndex = state.messagesJSON.map(m => m.id).indexOf(action.payload.id)
-  if (msgIndex > -1)
+  if (msgIndex > -1) {
```
[the remark this answers is not in the document]
Comment (vanbasten17) [reply packages/botonic-react/src/webchat/messages-reducer.js:41]: this should be ideal, I was thinking of it after having working with immer on the app 👍

File: packages/botonic-react/src/components/message.jsx [hunk 3]
```diff
@@ -97,6 +102,8 @@ export const Message = props => {
     timestamps = true,
     ...otherProps
   } = props
+  const isFromUser = () => from === 'user'
+  const isFromBot = () => from === 'bot'
```
Comment (ericmarcos) [inline packages/botonic-react/src/components/message.jsx:106]: Why declare this as a function and not simply a boolean variable?

File: packages/botonic-react/src/webchat/webchat.jsx [hunk 4]
```diff
@@ -271,6 +300,17 @@ export const Webchat = forwardRef((props, ref) => {
     webchatState.lastMessageUpdate,
   ])
 
+  useAsyncEffect(async () => {
+    if (!isOnline)
```
Comment (dpinol) [inline packages/botonic-react/src/webchat/webchat.jsx:304]: I would add here {} for symetry with else

**Pull request merged**